### PR TITLE
[IMP] web: Command palette

### PR DESCRIPTION
**Pull request by FrancoisGe**
The purpose of this commit is to:
- Improve the design of the command palette
- Add some data-command-category
- Scrolling through the command palette dropdown with the up/down arrows
should loop

Task-2590390

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/static/src/components/chatter_topbar/chatter_topbar.xml [hunk 1]
```diff
@@ -14,6 +14,8 @@
                             }"
                             t-att-disabled="chatter.isDisabled"
                             t-on-click="_onClickSendMessage"
+                            title="Send a message"
+                            data-hotkey="m"
```
Comment (brboi) [inline addons/mail/static/src/components/chatter_topbar/chatter_topbar.xml:18]: For all template changes in this PR, please check new coding style rules.

File: addons/mail/static/src/components/messaging_menu/messaging_menu.scss [hunk 2]
```diff
@@ -37,7 +37,7 @@
         * As a result, Min() is ignored by SCSS while CSS interprets as its min() function.
         */
         max-height: Min(calc(100vh - 140px), 630px);
-        z-index: 1100; // on top of chat windows
+        z-index: 1050; // on top of chat windows
```
Comment (brboi) [inline addons/mail/static/src/components/messaging_menu/messaging_menu.scss:40]: Has this been validated by a designer ?

File: addons/mail/static/src/webclient/commands/command_category.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (brboi) [inline addons/mail/static/src/webclient/commands/command_category.js:1]: Odoo module ?

**Review by brboi — COMMENTED** [review]
Looks good to me.

I left some comments.

Also, when I read the fix commit message, it suggests me that it is strictly related to Firefox. Which is not.

It would rather write something like:
`When hotkey overlays are removed, we now prevent default on keyup, as this may cause some browsers to display their menu bar (as the overlay key we use is ALT). e.g. Firefox.`

File: addons/web/static/src/legacy/xml/base.xml [hunk 4]
```diff
@@ -824,14 +831,14 @@
 
 <t t-name="ListView.buttons">
     <div class="o_list_buttons d-flex" role="toolbar" aria-label="Main actions">
-        <button type="button" class="btn btn-primary o_list_button_save" accesskey="s">
+        <button type="button" class="btn btn-primary o_list_button_save" title="Save record" accesskey="s">
```
Comment (ged-odoo) [inline addons/web/static/src/legacy/xml/base.xml:834]: shouldn't we take the textcontent and avoid the duplication?

File: addons/mail/static/src/webclient/commands/command_category.js (hunk 3 above)
Comment (ged-odoo) [reply addons/mail/static/src/webclient/commands/command_category.js:2]: having a webclient/ folder in mail sounds weird. anyway, this should be reviewed by the discuss team I guess @seb-odoo

File: addons/mail/static/src/webclient/commands/command_category.js [hunk 5]
```diff
@@ -1,7 +1,6 @@
 /** @odoo-module **/
 
 import { registry } from "@web/core/registry";
-import { _lt } from "@web/core/l10n/translation";
 
 const commandCategoryRegistry = registry.category("command_categories");
-commandCategoryRegistry.add("mail", { label: _lt("Discuss") }, { sequence: 20 });
+commandCategoryRegistry.add("mail", {}, { sequence: 20 });
```
Comment (ged-odoo) [inline addons/mail/static/src/webclient/commands/command_category.js:6]: it's very ugly...

File: addons/mail/static/src/webclient/commands/command_category.js (hunk 3 above)
Comment (seb-odoo) [reply addons/mail/static/src/webclient/commands/command_category.js:2]: Is it ready for us to be reviewed, or there are still changes to be applied?

File: addons/mail/static/src/widgets/form_renderer/form_renderer.js [hunk 6]
```diff
@@ -104,7 +104,7 @@ FormRenderer.include({
      * @returns {jQuery.Element}
      */
     _makeChatterContainerTarget() {
-        const $el = $('<div class="o_FormRenderer_chatterContainer"/>');
+        const $el = $('<div class="o_FormRenderer_chatterContainer" data-command-category="mail"/>');
```
Comment (seb-odoo) [inline addons/mail/static/src/widgets/form_renderer/form_renderer.js:107]: This is a compatibility layer, prefer moving this into `ChatterContainer` itself.

File: addons/mail/static/src/components/messaging_menu/messaging_menu.scss (hunk 2 above)
Comment (seb-odoo) [reply addons/mail/static/src/components/messaging_menu/messaging_menu.scss:40]: z-index principle is such a mess that it's hard to know for sure, but comment says "on top of chat windows", and chat window have z-index 1000, so this is ok to me.
If it was supposed to be above something else too, it should be been commented as well.

File: addons/mail/static/src/webclient/commands/command_category.js (hunk 3 above)
Comment (seb-odoo) [reply addons/mail/static/src/webclient/commands/command_category.js:2]: The guideline when overriding (or in this case registering) was (still is?) to use the same structure as the original code, so it looks ok to me.

I'm not against changing it either, but what better alternative do we have?

File: addons/mail/static/src/components/chatter_topbar/chatter_topbar.xml (hunk 1 above)
Comment (seb-odoo) [reply addons/mail/static/src/components/chatter_topbar/chatter_topbar.xml:18]: The new guideline is still under debate, so I would prefer if this followed the guideline that is applied on that file right now, which would be to have `title` and `data-hotkey` after `t-att-disable` but before `t-on-click`.

We'll do a pass on all our files to apply the new guideline later.

File: addons/web/static/tests/webclient/commands/command_service_tests.js [hunk 7]
```diff
@@ -39,8 +39,8 @@ QUnit.module("Command", {
         serviceRegistry.add("command", commandService);
 
         commandCategoryRegistry.add("custom-nolabel", {});
```
Comment (brboi) [inline addons/web/static/tests/webclient/commands/command_service_tests.js:41]: Remove this command, has none have labels now

File: addons/mail/static/src/webclient/commands/command_category.js (hunk 3 above)
Comment (FrancoisGe) [reply addons/mail/static/src/webclient/commands/command_category.js:2]: > Is it ready for us to be reviewed, or there are still changes to be applied?

Yes, it s ready.

**Review by seb-odoo — APPROVED** [review]

**Comment by ged-odoo** [comment]
robodoo r+ rebase-ff

**Pull request closed without merging**